Add test for using fuzziness parameter in multi_match query

cbuescher opened the pull request:
There was an issue with using fuzziness parameter in multi_match query that has been reported in #18710 and was fixed in Lucene 6.2 that is now used on master. In order to verify that fix and close the original issue this PR adds the test from that issue as an integration test.

Closes #18710

cbuescher committed:
Add test for using fuzziness parameter in multi_match query

There was an issue with using fuzziness parameter in multi_match query that has
been reported in #18710 and was fixed in Lucene 6.2 that is now used on master.
In order to verify that fix and close the original issue this PR adds the test
from that issue as an integration test.

nik9000 commented on core/src/test/java/org/elasticsearch/search/query/MultiMatchQueryIT.java:
```diff
@@ -635,6 +635,37 @@ public void testCrossFieldMode() throws ExecutionException, InterruptedException
     }
 
+    /**
+     * see https://github.com/elastic/elasticsearch/issues/18710
```
I'm fine with adding a link in the javadocs but could you add a short explanation so I don't have to go read the link? When I'm scanning javadocs there is 0% chance I'll go and read the link.

nik9000 commented on core/src/test/java/org/elasticsearch/search/query/MultiMatchQueryIT.java:
```diff
@@ -635,6 +635,37 @@ public void testCrossFieldMode() throws ExecutionException, InterruptedException
+        builders.add(client().prepareIndex(idx, "type", "2").setSource(
+                "title", "bar",
+                "body", "foo"));
+        indexRandom(true, false, builders);
```
If it is really important that these docs end up on separate shards can you add an assertion that they are on separate shards? That would explain the `true, false, builder` thing.

cbuescher commented:
@nik9000 thanks for the review, I added a hopefully better descrition to the test and also added checks that the two test documents live on different shards. This should communicate better that this test relies on this setup.

nik9000 commented:
Thanks for making the changes! Sorry I didn't notice that you'd made them.
